### PR TITLE
Update to syn v2

### PR DESCRIPTION
**Pull request by aumetra**
This PR updates `syn` to version 2.  
While I'm not 1:1 sure of the upgrade path between v1 and v2, this seems to pass the test suite just fine.

### CODE REVIEW

File: autometrics-macros/src/result_labels.rs [hunk 1]
```diff
@@ -94,21 +94,17 @@ fn conditional_label_clauses(
...
                         // Only lists are allowed
-                        let pair = match list.nested.first() {
-                            Some(syn::NestedMeta::Meta(syn::Meta::NameValue(pair))) => pair,
-                            _ => return Some(Err(Error::new_spanned(
-                            meta,
-                            format!("Only `{ATTR_LABEL}({RESULT_KEY} = \"RES\")` (RES can be {OK_KEY:?} or {ERROR_KEY:?}) is supported"),
-                            ))),
+                        let pair = match att.meta.require_list().and_then(|list| list.parse_args::<syn::MetaNameValue>()) {
+                            Ok(pair) => pair,
+                            Err(err) => return Some(Err(Error::new_spanned(&att.meta, err))),
```
Comment (mellowagain) [inline autometrics-macros/src/result_labels.rs:107]: is there a reason for changing the error message?

Comment (aumetra) [reply autometrics-macros/src/result_labels.rs]: Good catch! Forgot to change it back after some experiments

**Review by mellowagain — APPROVED** [review]

**Comment by mellowagain** [comment]
can you update the `CHANGELOG.md`? then it'll be good to go i think

**Comment by mellowagain** [comment]
thank you for your contribution!

**Pull request merged**